File: algorithms/math/bhaskara.py
"""
iagorrr ;)

bhaskara

delta = b^2 + 4ac

x = -b +- sqrt(detla)
        2a
"""
from math import sqrt

a, b, c = map(int, input().split())

delta = b**2 - 4*a*c
if a == 0 and b == 0 and c == 0:# infinite root. 
    print(-1)
elif a == 0 and b == 0:# no root.
    print(0)
elif a == 0: # one root.
    print(1)
    print(-c / b)
elif delta < 0: # no root.
    print(0)
elif delta == 0: # one root.
    print(1)
    x1 = (-b ) / (2*a)
    print(x1)
else: # two distinct root.
    x1 = (-b + sqrt(delta)) / (2*a)
    x2 = (-b - sqrt(delta))/ (2*a)
    print(2)
    if(x1 > x2):
        t = x1;
        x1 = x2
        x2 = t
    print(x1)
    print(x2)

File: algorithms/math/bhaskara.cpp
// iagorrr ;)
#include <bits/stdc++.h>
using namespace std;
#define fastio ios_base::sync_with_stdio(false); cin.tie(0); cout.tie(0);
using ll = long long;
using vll = vector<ll>;
using pll = pair<ll, ll>;
using vi = vector<int>;
using pii = pair<int, int>;
using ld = long double;
#define INV(xxxx) for(auto &xxx : xxxx) cin >> xxx;
#define all(a) a.begin(), a.end()
#define ss second
#define ff first
#define mp make_pair

void dbg_out() { cerr << endl; }
template <typename H, typename... T>
void dbg_out(H h, T... t) { cerr << ' ' << h; dbg_out(t...); }
#define dbg(...) { cerr << #__VA_ARGS__ << ':'; dbg_out(__VA_ARGS__); }

const ld EPS = 1e-5;
void run(){
  ll a, b, c; cin >> a >> b >> c;

  ld delta = (ld) b * b - 4.0*a*c;

  // infinite root count.
  if(a == 0 and b == 0 and c == 0) { // infinite root.
    cout << -1 << '\n';
  }
  else if (a == 0 and b == 0) {// no root.
    cout << 0 << '\n';
  }
  else if (a == 0) {
    cout << 1 << '\n';
    cout << ((ld) -c) / (b) << '\n';
  }
  else if (abs(delta) <= EPS) { // one root.
    cout << 1 << '\n';
    cout << ((ld)-b) / (2*a) << '\n';
  }
  else if (delta < 0) { // no root.
    cout << 0 << '\n';
  }
  else { // two distinct roots
    cout << 2 << '\n';
    ld x1 = ((ld) -b + (ld) sqrt(delta)) / (2.0*(ld)a);
    ld x2 = ((ld) -b - (ld) sqrt(delta)) / (2.0*(ld)a);
    if(x1 > x2) swap(x1, x2);
    cout << x1 << '\n' << x2 << '\n';
    
  }

}
int32_t main(void){ fastio;

  cout.precision(12);
  int t; t = 1;
  
  while(t--)
    run();
}

